app/services/hyrax/collections/collection_member_service.rb

```diff
@@ -81,7 +81,15 @@ def add_members_by_ids(collection:, new_member_ids:, user:)
         # @param new_members [Enumerable<Hyrax::Resource>] the new child collections and/or child works
         # @return [Enumerable<Hyrax::Resource>] updated member resources
         def add_members(collection:, new_members:, user:)
-          new_members.map { |new_member| add_member(collection: collection, new_member: new_member, user: user) }
+          messages = []
+          new_members.map do |new_member|
+            begin
+              add_member(collection: collection, new_member: new_member, user: user)
+            rescue Hyrax::SingleMembershipError => err
+              messages += [err.message]
+            end
```

Comment:
It was passing through the exception at the first member that violated single membership.  But the original code for adding members built up each violation message and passed back the unique messages.  This is updating adding multiple members at a time to do the same thing.